### PR TITLE
Add aml support

### PR DESCRIPTION
**Pull request by ayinloya**
This adds aml support

Test
- Build the gem locally `gem build`
- Install the gem `gem install --local smile-identity-core-2.2.0.gem`
- Add credentials to the `examples/aml_check.rb` file
- Run the example `ruby examples/aml_check.rb` for both **sandbox** and **prod**

### CODE REVIEW

File: lib/smile-identity-core/aml_check.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  # The AML Check product allows you to perform due diligence on your customers by screening them against
+  # global watchlists, politically exposed persons lists, and adverse media publications.
+  # For more info visit https://docs.smileidentity.com/products/for-individuals-kyc/aml-check
+  class AmlCheck
+    include Validations
+
+    ###
+    # Submit AML
+    # @param [Hash] partner_params the options to create a message with.
+    # @option opts [String] :job_type The job type, this should be 10
```
Comment (beaglebets) [inline lib/smile-identity-core/aml_check.rb]: job_type should be an integer

File: lib/smile-identity-core/aml_check.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      @url = if sid_server !~ URI::DEFAULT_PARSER.make_regexp
+               SmileIdentityCore::ENV::SID_SERVER_MAPPING[sid_server.to_s]
+             else
+               sid_server
+             end
+    end
+
+    # Submit AML
+    # @param [Hash] partner_params the options to create a job with.
+    # @option opts [String] :job_type The job type, this should be 10
```
Comment (beaglebets) [inline lib/smile-identity-core/aml_check.rb]: same

File: lib/smile-identity-core/aml_check.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    # @option opts [String] :job_id A unique value generated by you to track jobs on your end.
+    # @option opts [String] :user_id A unique value generated by you.
+    # @param [Hash] id_info
+    # @option opts [String] :countries An array that takes the customer’s known nationalities in 2-character format
```
Comment (beaglebets) [inline lib/smile-identity-core/aml_check.rb]: specifically ISO 3166

File: lib/smile-identity-core/aml_check.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    def submit_job(params)
+      @params = symbolize_keys(params)
+      @optional_info = @params['optional_info']
+      submit_requests
+    end
+
+    private
+
+    def symbolize_keys(params)
+      params.is_a?(Hash) ? params.transform_keys(&:to_sym) : params
```
Comment (beaglebets) [inline lib/smile-identity-core/aml_check.rb:52]: isn't this backwards?

File: lib/smile-identity-core/aml_check.rb [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+      request.on_complete do |response|
+        return response.body if response.success?
+
+        raise "#{response.code}: #{response.body}"
+      end
+      request.run
+    end
+
+    alias setup_requests submit_requests
```
Comment (beaglebets) [inline lib/smile-identity-core/aml_check.rb]: why this? just call the method what it should be.

**Review by beaglebets — CHANGES_REQUESTED** [review]

File: lib/smile-identity-core/aml_check.rb (hunk 4 above)
Comment (ayinloya) [reply lib/smile-identity-core/aml_check.rb:52]: We allow hash keys to be transformed to sym but if it is not a hash, we return it as is

File: lib/smile-identity-core/aml_check.rb (hunk 5 above)
Comment (ayinloya) [reply lib/smile-identity-core/aml_check.rb]: Will update this

**Review by beaglebets — DISMISSED** [review]

**Comment by beaglebets** [comment]
I'll reapprove once you get tests passing again, presumably just need to change the name of the method to be the name you were aliasing to

**Comment by ayinloya** [comment]
> I'll reapprove once you get tests passing again, presumably just need to change the name of the method to be the name you were aliasing to

Thanks, resolved with some few updates

**Review by beaglebets — APPROVED** [review]

**Pull request merged**